android/kotlin/v2/AmpliApp/app/src/main/java/com/amplitude/ampli/IngestionMetadataPlugin.kt

```diff
@@ -0,0 +1,20 @@
+package com.amplitude.ampli
+
+import com.amplitude.core.events.BaseEvent
+import com.amplitude.core.platform.Plugin
+
+class IngestionMetadataPlugin(ingestionMetadata: IngestionMetadata) : Plugin {
+    override val type: Plugin.Type = Plugin.Type.Enrichment
+    override lateinit var amplitude: com.amplitude.core.Amplitude
+    val ingestionMetadata = ingestionMetadata
+
+    override fun execute(event: BaseEvent): BaseEvent? {
+        val amplitudeExtra = (event.extra?.get("amplitude") ?: mapOf<String, Any>()) as Map<String, Any>
+        val ingestionMetadataMap = mapOf(
+            "ingestionMetadata" to  ingestionMetadata.toJsonObject()
```

Comment:
Do we really need this to be a JSONObject? Could it be just a map?

Reply:
we already has jsonObject <-> ingestionMetadata converter in core sdk. 
If we use map instead, do we also need to have a converter to convert the map to ingestionMetadata?